### ISSUE TITLE
Fails creating OSDs on servers with multiple drives

### ISSUE THREAD

**Issue by dimm0** [rook/rook#1480]
Is this a bug report or feature request?
* Bug Report

**Bug Report**

What happened:
I installed rook on one node with ~30 drives, and it failed on 16th. I did not record the error exactly, but it was complaining about hitting the limit of fs.aio-max-nr, which is equal to 65536 in CentOS 7.
After that the OSD pod restarted and was never able to recover, complaining that the OSD folder already exists.

What you expected to happen:
rook should set this limit itself, or give a clear error and continue initializing OSDs after POD crash

How to reproduce it (minimal and precise):
install minikube, init rook on system with multiple drives

**Environment**:
* OS (e.g. from /etc/os-release):
CentOS Linux 7
* Kernel (e.g. `uname -a`):
Linux sdsu-ceph-01.sdsu.edu 4.14.15-1.el7.elrepo.x86_64 #1 SMP Tue Jan 23 20:28:26 EST 2018 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration:
baremetal
* Rook version (use `rook version` inside of a Rook Pod):
v0.6.0-207.g516e96f
* Kubernetes version (use `kubectl version`):
1.9.0
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
minikube

**Comment by jbw976** [comment]
We currently determine if an OSD is "created" by checking for the `whoami` file.  If it doesn't exist then we assume the OSD is brand new and we go through the OSD mkfs process.  It looks like in this scenario, the OSD mkfs failed before the `whoami` file was created, so when we retried we ran mkfs again but it failed because there were some existing artifacts already present.

One way to address this would be if the whoami file does not exist, to delete any other files/dirs that may be present.  The mkfs did not complete and it's in an intermediate state, so we should clean it up and start fresh.
